kucher/view/main_window/register_view_widget/dialog_export.py

```diff
@@ -0,0 +1,197 @@
+
+from kucher.view.device_model_representation import Register
+from kucher.view.utils import show_error, lay_out_horizontally, lay_out_vertically, get_monospace_font
+
+_logger = getLogger(__name__)
+
+
+class ExportDialogWindow(QWidget):
+
+    def __init__(self, registers):
+        self.registers = registers
+        super().__init__()
+        self.setGeometry(10, 10, 1000, 700)
+        self._saveFileDialog()
+
+    def _saveFileDialog(self):
+        self.fileName, _ = QFileDialog.getSaveFileName(self, 'Export configuration file', 'config.yml',
+                                                       'YAML Files (*.yml  *.yaml);;All Files (*)')
+        if not self.fileName:
+            return
+        try:
+            _file = open(self.fileName, 'w+')
+
+        except IOError:
+            _logger.exception(f'File {self.fileName} could not be open')
+            return
+
+        async def executor():
+            register_yaml = {}
+            for reg in self.registers:
+                if reg.mutable:
+                    try:
+                        register_yaml[reg.name] = await reg.read_through()
+                    except Exception as ex:
+                        _logger.exception(f'Register {reg.name} could not be read')
+
+            yaml.dump(register_yaml, _file)
+            _file.close()
+
+        asyncio.get_event_loop().create_task(executor())
+
+
+class ImportDialogWindow(QWidget):
+
+    def __init__(self, parent, registers):
+        self.parent = parent
+        self.registers = registers
+        super().__init__()
+        self.setGeometry(10, 10, 1000, 700)
+        self._saveFileDialog()
+
+    def _saveFileDialog(self):
+        self.fileName, _ = QFileDialog.getOpenFileName(self, 'Import configuration file', '',
+                                                       'YAML Files (*.yml *.yaml);;All Files (*)')
+        if not self.fileName:
+            return
+        try:
+            self._file = open(self.fileName, 'r')
+        except IOError:
+            _logger.exception(f'File {self.fileName} could not be open')
+            return
+
+        # Check before writing any register if they are available on the device and within the permitted range
+        self.imported_registers = yaml.safe_load(self._file)
+        try:
+            for reg_check in self.registers:
+                if reg_check.name in self.imported_registers.keys():
+                    if not reg_check.mutable:
+                        show_error('Import failed',
+                                   f'Cannot import {self.fileName}',
+                                   f'This parameter cannot be modified on this device:\n{reg_check.name}',
+                                   self.parent)
+                        self._file.close()
+                        _logger.exception(f'Import failed : This parameter cannot be modified on this '
+                                          f'device: {reg_check.name}')
```

Comment:
It's best to log the parameter together with its metadata. There are other occurrences where this comment applies.
```suggestion
                                          f'device: {reg_check}')
```